src/main/java/hudson/plugins/active_directory/LDAPSearchBuilder.java

```diff
@@ -70,11 +70,24 @@ public LDAPSearchBuilder returns(String... attributes) {
         return this;
     }
 
+    /**
+     * Add all the attributes returned by the search
+     *
+     * @param filter
+     * @param args
+     *
+     * @return Return all the attributes of the search and adds the DN correctly formatted per RFC 2253
+     */
     public Attributes searchOne(String filter, Object... args) throws NamingException {
         NamingEnumeration<SearchResult> r = search(filter,args);
         try {
             if (r.hasMore()) {
-                Attributes attrs = r.next().getAttributes();
+                SearchResult searchResult = r.next();
+                Attributes attrs = searchResult.getAttributes();
+                //We need to use getNameInNamespace in order to correctly format everything
+                //to be able to use LdapName later and get the DN correctly formatted with escaped
+                //characters like slash
+                attrs.put("distinguishedNameFormatted", searchResult.getNameInNamespace());
```

Comment:
the naming makes this look a bit official - also as you use the same string elsewhere this should be a package scoped final variable.